moto/s3/responses.py

```diff
@@ -1217,7 +1217,7 @@ def _key_response_get(self, bucket_name, query, key_name, headers):
             if_unmodified_since = str_to_rfc_1123_datetime(if_unmodified_since)
             if key.last_modified > if_unmodified_since:
                 raise PreconditionFailed("If-Unmodified-Since")
-        if if_match and key.etag != if_match:
+        if if_match and key.etag != '"{0}"'.format(if_match):
```

Comment:
Thanks for this PR @macnev2013 👍 

Wondering if we should extend the `if` check, to cover both cases for backwards compatibility (with and without the double quotes):
```
         if if_match and key.etag not in [if_match, '"{0}"'.format(if_match)]:
             ...
```
Wdyt?

Reply:
I have checked AWS behavior, but it always contains "etag" in double quotes. Ideally we should consider "etag" and etag as different values.

Reply:
Yep, that's understood.

However, I was more thinking of cases where internally moto is already stripping off the quotes from the etag (already today, or potentially in the future). E.g. here: https://github.com/spulec/moto/blob/master/moto/s3/models.py#L318-L319

But happy to merge as is for now - we may need to change later if and when we try to make the moto tests pass here.

Reply:
Yep make sense. Lets make that change and make it more flexible for future.